core/src/main/java/org/jdbi/v3/core/array/SqlArrayArgumentFactory.java

```diff
@@ -47,8 +47,12 @@
                     .map(arrayType -> new SqlArrayArgument<>(arrayType, value));
         }
 
-        return GenericTypes.findGenericParameter(type, Collection.class)
-                .flatMap(lookup)
+        final Collection<?> collection = (Collection<?>) value;
+        // If there's no static type, guess based on the first element
+        final Type eltType = GenericTypes.findGenericParameter(type, Collection.class)
+            .orElse(collection.isEmpty() ? String.class : collection.iterator().next().getClass());
```

Comment:
I'm OK with requiring explicit type binding like you suggest above, the convenience methods make it a lot nicer.  I don't think the `String` default is particularly harmful since it is only used if there are no elements, in which case no elements are bound, but I agree it looks weird.